lib/forwarder.ts

```diff
@@ -13,18 +13,27 @@ import * as crypto from "crypto";
 import { LambdaDestination } from "@aws-cdk/aws-logs-destinations";
 const SUBSCRIPTION_FILTER_PREFIX = "DatadogSubscriptionFilter";
 export function addForwarder(scope: cdk.Construct, lambdaFunctions: lambda.Function[], forwarderARN: string) {
-  const forwarder = lambda.Function.fromFunctionArn(scope, "forwarder", forwarderARN);
+  const forwarderConstructID = "forwarder" + crypto.createHash("sha256").update(forwarderARN).digest("hex");
+  let forwarder;
+  if (scope.node.tryFindChild(forwarderConstructID)) {
+    forwarder = scope.node.tryFindChild(forwarderConstructID) as lambda.IFunction;
+  } else {
+    forwarder = lambda.Function.fromFunctionArn(scope, forwarderConstructID, forwarderARN);
+  }
   const forwarderDestination = new LambdaDestination(forwarder);
   lambdaFunctions.forEach((lam) => {
-    const subscriptionFilterValue: string = crypto.createHash("sha256").update(lam.functionArn).digest("hex");
+    const subscriptionFilterValue: string = crypto
```

Comment:
I would suggest pulling this logic out into a separate helper function, maybe `generateSubscriptionFilterName(functionArn, forwarderArn)`. Then the logic for `addForwarder()` itself becomes simpler.

Reply:
Two smaller functions is pretty much always better than one big function with lots of details in my book.